### ISSUE TITLE
Compilation error on GHC 9.2.2 (pre-release): Couldn't match expected type ‘Word#’ with actual type ‘Word8#’

### ISSUE THREAD

**Issue by maerten** [vincenthz/hs-memory#90]
```
memory               > configure
memory               > Configuring memory-0.16.0...
memory               > build
memory               > Preprocessing library for memory-0.16.0..
memory               > Building library for memory-0.16.0..
memory               > [ 1 of 28] Compiling Data.Memory.ExtendedWords
memory               > [ 2 of 28] Compiling Data.Memory.Internal.Compat
memory               > [ 3 of 28] Compiling Data.Memory.Endian
memory               > [ 4 of 28] Compiling Data.Memory.Hash.SipHash
memory               > [ 5 of 28] Compiling Data.Memory.Encoding.Base16
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:46:49: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘convertByte’, namely ‘a’
memory               >       In the expression: convertByte a
memory               >       In a pattern binding: !(# w1, w2 #) = convertByte a
memory               >    |
memory               > 46 |                     !(# w1, w2 #) = convertByte a
memory               >    |                                                 ^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:47:49: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘convertByte’, namely ‘b’
memory               >       In the expression: convertByte b
memory               >       In a pattern binding: !(# w3, w4 #) = convertByte b
memory               >    |
memory               > 47 |                     !(# w3, w4 #) = convertByte b
memory               >    |                                                 ^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:48:49: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘convertByte’, namely ‘c’
memory               >       In the expression: convertByte c
memory               >       In a pattern binding: !(# w5, w6 #) = convertByte c
memory               >    |
memory               > 48 |                     !(# w5, w6 #) = convertByte c
memory               >    |                                                 ^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:49:49: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘convertByte’, namely ‘d’
memory               >       In the expression: convertByte d
memory               >       In a pattern binding: !(# w7, w8 #) = convertByte d
memory               >    |
memory               > 49 |                     !(# w7, w8 #) = convertByte d
memory               >    |                                                 ^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:58:49: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘convertByte’, namely ‘b’
memory               >       In the expression: convertByte b
memory               >       In a pattern binding: !(# w1, w2 #) = convertByte b
memory               >    |
memory               > 58 |                     !(# w1, w2 #) = convertByte b
memory               >    |                                                 ^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:85:49: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘convertByte’, namely ‘w’
memory               >       In the expression: convertByte w
memory               >       In a pattern binding: !(# w1, w2 #) = convertByte w
memory               >    |
memory               > 85 |                 let !(# w1, w2 #) = convertByte w
memory               >    |                                                 ^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:86:51: error:
memory               >     • Couldn't match expected type ‘Word8#’ with actual type ‘Word#’
memory               >     • In the first argument of ‘W8#’, namely ‘w1’
memory               >       In the third argument of ‘pokeByteOff’, namely ‘(W8# w1)’
memory               >       In a stmt of a 'do' block: pokeByteOff bout (i * 2) (W8# w1)
memory               >    |
memory               > 86 |                 pokeByteOff bout (i * 2)     (W8# w1)
memory               >    |                                                   ^^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:87:51: error:
memory               >     • Couldn't match expected type ‘Word8#’ with actual type ‘Word#’
memory               >     • In the first argument of ‘W8#’, namely ‘w2’
memory               >       In the third argument of ‘pokeByteOff’, namely ‘(W8# w2)’
memory               >       In a stmt of a 'do' block: pokeByteOff bout (i * 2 + 1) (W8# w2)
memory               >    |
memory               > 87 |                 pokeByteOff bout (i * 2 + 1) (W8# w2)
memory               >    |                                                   ^^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:96:25: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the expression: indexWord8OffAddr# table (word2Int# index)
memory               >       In an equation for ‘r’:
memory               >           r table index = indexWord8OffAddr# table (word2Int# index)
memory               >       In an equation for ‘convertByte’:
memory               >           convertByte b
memory               >             = (# r tableHi b, r tableLo b #)
memory               >             where
memory               >                 r :: Addr# -> Word# -> Word#
memory               >                 r table index = indexWord8OffAddr# table (word2Int# index)
memory               >                 !tableLo
memory               >                   = "0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef\
memory               >             \0123456789abcdef0123456789abcdef"#
memory               >                 !tableHi
memory               >                   = "00000000000000001111111111111111\
memory               >             \22222222222222223333333333333333\
memory               >             \44444444444444445555555555555555\
memory               >             \66666666666666667777777777777777\
memory               >             \88888888888888889999999999999999\
memory               >             \aaaaaaaaaaaaaaaabbbbbbbbbbbbbbbb\
memory               >             \ccccccccccccccccdddddddddddddddd\
memory               >             \eeeeeeeeeeeeeeeeffffffffffffffff"#
memory               >    |
memory               > 96 |         r table index = indexWord8OffAddr# table (word2Int# index)
memory               >    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:134:70: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘word2Int#’, namely ‘index’
memory               >       In the second argument of ‘indexWord8OffAddr#’, namely
memory               >         ‘(word2Int# index)’
memory               >       In the first argument of ‘W8#’, namely
memory               >         ‘(indexWord8OffAddr# tableLo (word2Int# index))’
memory               >     |
memory               > 134 |         rLo (W8# index) = W8# (indexWord8OffAddr# tableLo (word2Int# index))
memory               >     |                                                                      ^^^^^
memory               >
memory               > /private/var/folders/0c/zmpjp7l568xcvnt49pkkn5p00000gn/T/stack-a625ece36a2e1acb/memory-0.16.0/Data/Memory/Encoding/Base16.hs:135:70: error:
memory               >     • Couldn't match expected type ‘Word#’ with actual type ‘Word8#’
memory               >     • In the first argument of ‘word2Int#’, namely ‘index’
memory               >       In the second argument of ‘indexWord8OffAddr#’, namely
memory               >         ‘(word2Int# index)’
memory               >       In the first argument of ‘W8#’, namely
memory               >         ‘(indexWord8OffAddr# tableHi (word2Int# index))’
memory               >     |
memory               > 135 |         rHi (W8# index) = W8# (indexWord8OffAddr# tableHi (word2Int# index))
memory               >     |                                                                      ^^^^^
```

**Comment by maerten** [comment]
Using the branch mentioned in https://github.com/vincenthz/hs-memory/pull/87 fixes the issue for me.

**Comment by ysangkok** [comment]
@maerten This can be closed, a release with GHC 9.2 support was made.
